packages/augur-state/src/api/Users.ts

```diff
@@ -1,30 +1,222 @@
+  averagePrice: string; // denominated in tokens/share. average price user paid for shares in the current open position
+  realized: string; // realized profit in tokens (eg. ETH) user already got from this market outcome. "realized" means the user bought/sold shares in such a way that the profit is already in the user's wallet
+  unrealized: string; // unrealized profit in tokens (eg. ETH) user could get from this market outcome. "unrealized" means the profit isn't in the user's wallet yet; the user could close the position to "realize" the profit, but instead is holding onto the shares. Computed using last trade price.
+  total: string; // total profit in tokens (eg. ETH). Always equal to realized + unrealized
+  unrealizedCost: string; // denominated in tokens. Cost of shares in netPosition
+  realizedCost: string; // denominated in tokens. Cumulative cost of shares included in realized profit
+  totalCost: string; // denominated in tokens. Always equal to unrealizedCost + realizedCost
+  realizedPercent: string; // realized profit percent (ie. profit/cost)
+  unrealizedPercent: string; // unrealized profit percent (ie. profit/cost)
+  totalPercent: string; // total profit percent (ie. profit/cost)
+  currentValue: string; // current value of netPosition, always equal to unrealized minus frozenFunds
+}
+
+export interface UserTradingPositions {
+  tradingPositions: Array<TradingPosition>; // per-outcome TradingPosition, where unrealized profit is relative to an outcome's last price (as traded by anyone)
+  tradingPositionsPerMarket: { // per-market rollup of trading positions
+    [marketId: string]: MarketTradingPosition,
+  };
+  frozenFundsTotal: string; // User's total frozen funds. See docs on FrozenFunds. This total includes market validity bonds in addition to sum of frozen funds for all market outcomes in which user has a position.
 }
 
-export class Users {
-  constructor() {
+export class Users<TBigNumber> {
+  private readonly db: DB<TBigNumber>;
+
+  constructor(db: DB<TBigNumber>) {
+    this.db = db;
   }
 
-  public async getUserTradingPositions(params: GetUserTradingPositionsParams): Promise<void> {
-    // TODO
+  public async getUserTradingPositions(params: GetUserTradingPositionsParams): Promise<UserTradingPositions> {
+    if (!params.account) {
+      throw new Error("'getTradingHistory' requires an 'account' param be provided");
+    }
+    if (!params.universe && !params.marketId) {
+      throw new Error("'getTradingHistory' requires a 'universe' or 'marketId' param be provided");
+    }
+    const request = {
+      selector: {
+        universe: params.universe,
+        market: params.marketId,
+        account: params.account,
+      },
+      sort: params.sortBy ? [params.sortBy] : undefined,
+      limit: params.limit,
```

Comment:
`limit` & `offset` are optional, so I think you'll need a check for these as well, similar to the line above.

Reply:
If they arent included this will just not use them

Reply:
To clarify if the value is undefined they wont apply to the find request